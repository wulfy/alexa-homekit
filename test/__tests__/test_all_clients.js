const base_config = require("../config_tests/base_config");
const {override} = require("../config_tests/functions_override");

let file = '';
let domoticz_mockup = '';
describe('Testing all clients', () => { 
    for(let i = 10; i < 13; i++){
    	file = "../mockups/client"+i+"Mockup";
		domoticz_mockup = { DOMOTICZ_GET_DEVICES } = require(file);
		override(domoticz_mockup.DOMOTICZ_GET_DEVICES);

		it('DISCOVERY TESTING client' + i, done => {
			let context = {};
			context.succeed = function (data){
				const testData = JSON.stringify(data); 
				expect(testData).toMatchSnapshot();
				done();
			};
			base_config.handler(base_config.mockups.ALEXA_DISCOVERY_REQUEST_EXAMPLE,context);
			console.log("testing");
		});
	}
});