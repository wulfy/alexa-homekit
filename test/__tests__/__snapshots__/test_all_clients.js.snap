// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing all clients DISCOVERY TESTING client10 1`] = `"{\\"event\\":{\\"header\\":{\\"namespace\\":\\"Alexa.Discovery\\",\\"name\\":\\"Discover.Response\\",\\"payloadVersion\\":\\"3\\",\\"messageId\\":\\"abc-123-def-456\\"},\\"payload\\":{\\"endpoints\\":[{\\"endpointId\\":\\"2_RFXCOM_Blinds\\",\\"manufacturerName\\":\\"RFXCOM\\",\\"friendlyName\\":\\"GVsalon\\",\\"description\\":\\"(pas de description)\\",\\"displayCategories\\":[\\"LIGHT\\"],\\"cookie\\":{\\"MaxDimLevel\\":\\"0\\"},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.PercentageController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"percentage\\",\\"value\\":\\"0\\"}],\\"retrievable\\":true}},{\\"interface\\":\\"Alexa.PowerController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"powerState\\",\\"value\\":\\"()=> ('Closed' === 'Off' || 'Closed' === 'Closed') ? 'OFF' : 'ON'\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}}]},{\\"endpointId\\":\\"3_RFXCOM_Blinds\\",\\"manufacturerName\\":\\"RFXCOM\\",\\"friendlyName\\":\\"Vsalon\\",\\"description\\":\\"(pas de description)\\",\\"displayCategories\\":[\\"LIGHT\\"],\\"cookie\\":{\\"MaxDimLevel\\":\\"0\\"},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.PercentageController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"percentage\\",\\"value\\":\\"0\\"}],\\"retrievable\\":true}},{\\"interface\\":\\"Alexa.PowerController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"powerState\\",\\"value\\":\\"()=> ('Closed' === 'Off' || 'Closed' === 'Closed') ? 'OFF' : 'ON'\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}}]},{\\"endpointId\\":\\"4_RFXCOM_Blinds\\",\\"manufacturerName\\":\\"RFXCOM\\",\\"friendlyName\\":\\"VCuisine\\",\\"description\\":\\"(pas de description)\\",\\"displayCategories\\":[\\"LIGHT\\"],\\"cookie\\":{\\"MaxDimLevel\\":\\"0\\"},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.PercentageController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"percentage\\",\\"value\\":\\"0\\"}],\\"retrievable\\":true}},{\\"interface\\":\\"Alexa.PowerController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"powerState\\",\\"value\\":\\"()=> ('Closed' === 'Off' || 'Closed' === 'Closed') ? 'OFF' : 'ON'\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}}]},{\\"endpointId\\":\\"5_RFXCOM_Blinds\\",\\"manufacturerName\\":\\"RFXCOM\\",\\"friendlyName\\":\\"VSalleamanger\\",\\"description\\":\\"(pas de description)\\",\\"displayCategories\\":[\\"LIGHT\\"],\\"cookie\\":{\\"MaxDimLevel\\":\\"0\\"},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.PercentageController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"percentage\\",\\"value\\":\\"0\\"}],\\"retrievable\\":true}},{\\"interface\\":\\"Alexa.PowerController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"powerState\\",\\"value\\":\\"()=> ('Closed' === 'Off' || 'Closed' === 'Closed') ? 'OFF' : 'ON'\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}}]},{\\"endpointId\\":\\"6_RFXCOM_Blinds\\",\\"manufacturerName\\":\\"RFXCOM\\",\\"friendlyName\\":\\"VBureau\\",\\"description\\":\\"(pas de description)\\",\\"displayCategories\\":[\\"LIGHT\\"],\\"cookie\\":{\\"MaxDimLevel\\":\\"0\\"},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.PercentageController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"percentage\\",\\"value\\":\\"0\\"}],\\"retrievable\\":true}},{\\"interface\\":\\"Alexa.PowerController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"powerState\\",\\"value\\":\\"()=> ('Closed' === 'Off' || 'Closed' === 'Closed') ? 'OFF' : 'ON'\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}}]}]}}}"`;
